onnx/test/shape_inference_test.py

```diff
@@ -996,6 +996,20 @@ def test_maxpool_with_dilations(self):  # type: () -> None
             [])
         self._assert_inferred(graph, [make_tensor_value_info("Y", TensorProto.FLOAT, (5, 3, 2, 2))])
 
+    def test_maxpool_with_same_upper_padding_and_stride(self):  # type: () -> None
```

Comment:
Maybe add a few more tests ? 
* NOTSET case
* strides = 1 with either SAME_UPPER or SAME_LOWER (kind of basic, but good to have)

Reply:
NOTSET is covered by old tests already. I added one more test for strides = 1